modules/generator/instance.go

```diff
@@ -113,60 +116,92 @@ func (i *instance) watchOverrides() {
 	}
 }
 
-func (i *instance) updateProcessors(desiredProcessors map[string]struct{}) error {
+func (i *instance) updateProcessors() error {
+	desiredProcessors := i.overrides.MetricsGeneratorProcessors(i.instanceID)
+	desiredCfg := i.cfg.Processor.copyWithOverrides(i.overrides, i.instanceID)
+
 	i.processorsMtx.RLock()
-	toAdd, toRemove := i.diffProcessors(desiredProcessors)
+	toAdd, toRemove, toReplace, err := i.diffProcessors(desiredProcessors, desiredCfg)
 	i.processorsMtx.RUnlock()
 
-	if len(toAdd) == 0 && len(toRemove) == 0 {
+	if err != nil {
+		return err
+	}
+	if len(toAdd) == 0 && len(toRemove) == 0 && len(toReplace) == 0 {
 		return nil
 	}
 
 	i.processorsMtx.Lock()
 	defer i.processorsMtx.Unlock()
 
 	for _, processorName := range toAdd {
-		err := i.addProcessor(processorName)
+		err := i.addProcessor(processorName, desiredCfg)
 		if err != nil {
 			return err
 		}
 	}
 	for _, processorName := range toRemove {
 		i.removeProcessor(processorName)
 	}
+	for _, processorName := range toReplace {
+		i.removeProcessor(processorName)
+
+		err := i.addProcessor(processorName, desiredCfg)
```

Comment:
i feel like this is a little dangerous b/c we could remove a tenants processor and then fail to add it back during an overrides update. as long as we have sufficient alerts around `metricActiveProcessorsUpdateFailed` it should be ok?

Reply:
Yeah, that's definitely a concern. This can already happen btw if you have a typo in your processor name.
I.e. the following config will fail because the processor name is incorrect:
```diff
        metrics_generator_processors:
            - servicegraphs
```
We have an alert on `tempo_metrics_generator_active_processors_update_failed_total` to catch this 🙂